Bugsnag/KSCrash/Source/KSCrash/Recording/Sentry/BSG_KSCrashSentry_Signal.c

```diff
@@ -126,15 +137,19 @@ void bsg_kssighndl_i_handleSignal(int sigNum, siginfo_t *signalInfo,
 
     BSG_KSLOG_DEBUG(
         "Re-raising or chaining signal for regular handlers to catch.");
-    struct sigaction previous = bsg_g_previousSignalHandlers[sigNum];
-    if (previous.sa_flags & SA_SIGINFO) {
-        previous.sa_sigaction(sigNum, signalInfo, userContext);
-    } else if (previous.sa_handler == SIG_DFL) {
+    struct sigaction *previous = get_previous_sigaction(sigNum);
+    if(previous == NULL) {
+        BSG_KSLOG_ERROR("BUG: Could not find handler for signal %d", sigNum);
```

Comment:
Why is it a bug if we couldn't find the previous handler?

Reply:
We fill out the entire table when we start up, so this could only happen if we somehow got a signal number we weren't supposed to. Technically this should be impossible, but I'd rather put in a log msg :)